kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/metrics/MetricsCollector.java

```diff
@@ -235,31 +236,24 @@ public MetricsCollector.Builder toBuilder() {
      * @param builder the builder
      */
     protected MetricsCollector(Builder builder) {
-        if (builder.namespaceName == null || builder.namespaceName.isEmpty())
-            builder.namespaceName = kubeClient().getNamespace();
-        if (builder.scraperPodName == null || builder.scraperPodName.isEmpty())
-            throw new InvalidParameterException("Scraper Pod name is not set");
-        if (builder.componentType == null)
-            throw new InvalidParameterException("Component type is not set");
-
-        componentType = builder.componentType;
+        Objects.requireNonNull(builder.componentType, "Component type not set");
 
-        if (builder.metricsPort <= 0)
-            builder.metricsPort = getDefaultMetricsPort();
-        if (builder.metricsPath == null || builder.metricsPath.isEmpty())
-            builder.metricsPath = getDefaultMetricsPath();
+        if (Optional.ofNullable(builder.scraperPodName).isEmpty()) {
+            throw new InvalidParameterException("Scraper Pod name is not set");
+        }
 
-        namespaceName = builder.namespaceName;
         scraperPodName = builder.scraperPodName;
-        metricsPort = builder.metricsPort;
-        metricsPath = builder.metricsPath;
+        namespaceName = (Optional.ofNullable(builder.namespaceName).isEmpty()) ? kubeClient().getNamespace() : builder.namespaceName;
```

Comment:
```suggestion
        namespaceName = Optional.ofNullable(builder.namespaceName).orElse(kubeClient().getNamespace());
```

Reply:
`orElse` is also checking if the string is empty? or only if it is null?